psi4/src/psi4/libfock/cubature.cc

```diff
@@ -4309,18 +4318,33 @@ void DFTGrid::buildGridFromOptions(std::map<std::string, int> int_opts_map,
         // printf("%s : %d\n", key.c_str(), full_int_options[key]);
     }
 
+    std::map<std::string, double> full_float_options;
+    std::vector<std::string> float_keys = {"DFT_BS_RADIUS_ALPHA",
+                                           "DFT_PRUNING_ALPHA",
+                                           "DFT_WEIGHTS_TOLERANCE",
+                                           "DFT_BLOCK_MAX_RADIUS",
+                                           "DFT_BASIS_TOLERANCE"};
+    for (auto key : float_keys) {
+        if (float_opts_map.find(key) != float_opts_map.end()) {
+            full_float_options[key] = float_opts_map[key];
+        } else {
+            full_float_options[key] = options_.get_double(key);
+        }
+        // prfloatf("%s : %d\n", key.c_str(), full_float_options[key]);
+    }
+
     // TODO: Move both the double and bool parts from a global to a grid specific option.
     //  possibly via an approach listed here:
     //  https://stackoverflow.com/questions/24702235/c-stdmap-holding-any-type-of-value/24702596
```

Comment:
std::any from C++17 following this discussion would be a suggestion how to generalize the option map.